presto-main/src/main/java/com/facebook/presto/cost/CachingCostProvider.java

```diff
@@ -54,6 +55,10 @@ public CachingCostProvider(CostCalculator costCalculator, StatsProvider statsPro
     @Override
     public PlanNodeCostEstimate getCumulativeCost(PlanNode node)
     {
+        if (!isEnableStatsCalculator(session)) {
```

Comment:
is this basically a shortcut so you don't keep trying to calculate cost and coming up empty because all stats are unknown?

Reply:
> is this basically a shortcut so you don't keep trying to calculate cost and coming up empty because all stats are unknown?

I'm really being paranoid here. Just want to make sure that this part of functionality is really killed for good.

Reply:
I extracted it into a separate commit